json-path/src/main/java/com/jayway/jsonpath/internal/token/PathToken.java

```diff
@@ -40,7 +45,13 @@ void handleObjectProperty(String currentPath, Object model, EvaluationContextImp
 
         if(properties.size() == 1) {
             String property = properties.get(0);
-            String evalPath = currentPath + "['" + property + "']";
+        	StringBuilder buf = BUFFERS.get() ;
+        	buf.setLength(0);
+        	buf.append(currentPath) ;
+        	buf.append('[') ;
```

Comment:
I have committed a fix.

Let me know in case of other problem.

Steph

2015-03-19 16:12 GMT+01:00 kallestenflo <notifications@github.com>:

> In
> json-path/src/main/java/com/jayway/jsonpath/internal/token/PathToken.java
> <https://github.com/jayway/JsonPath/pull/71#discussion_r26763076>:
>
> > @@ -40,7 +45,13 @@ void handleObjectProperty(String currentPath, Object model, EvaluationContextImp
> >
> >          if(properties.size() == 1) {
> >              String property = properties.get(0);
> > -            String evalPath = currentPath + "['" + property + "']";
> > +        	StringBuilder buf = BUFFERS.get() ;
> > +        	buf.setLength(0);
> > +        	buf.append(currentPath) ;
> > +        	buf.append('[') ;
>
> Tests break by this commit.
>
> You have missed the ' char wrapping the currentPath inside the brackets.
> It should probaly be:
>
> buf.append('[');
> buf.append('\'');
> buf.append(currentPath);
> buf.append('\'');
> buf.append(']');...
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/jayway/JsonPath/pull/71/files#r26763076>.
>